packages/apps/stock/src/features/stockMoveLineSlice.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {createAsyncThunk, createSlice} from '@reduxjs/toolkit';
+import {handlerApiCall} from '@axelor/aos-mobile-core';
+import {
+  addTrackingNumber as _addTrackingNumber,
+  fetchStockMoveLine as _fetchStockMoveLine,
+} from '../api';
+
+export const addTrackingNumber = createAsyncThunk(
+  'stockMoveLine/addTrackingNumber',
+  async function (data = {}, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _addTrackingNumber,
+      data,
+      action: 'add tracking number to stock move line',
+      getState,
+      responseOptions: {showToast: true},
+    }).then(res => {
+      return handlerApiCall({
+        fetchFunction: _fetchStockMoveLine,
+        data: {stockMoveLineId: res?.id},
+        action: 'fetch stock move line',
+        getState,
+        responseOptions: {isArrayResponse: false},
+      });
+    });
+  },
+);
+
+export const fetchStockMoveLine = createAsyncThunk(
+  'stockMoveLine/fetchStockMoveLine',
+  async function (data = {}, {getState}) {
+    return handlerApiCall({
+      fetchFunction: _fetchStockMoveLine,
+      data,
+      action: 'fetch stock move line',
+      getState,
+      responseOptions: {isArrayResponse: false},
+    });
+  },
+);
+
+const initialState = {
+  loading: false,
+  stockMoveLine: {},
+};
```

Comment:
Having a slice for all stockMoveLine types is not really a good pratice as it can trigger unwanted updates of hooks.
For example, if you are updating customer delivery line, it can also trigger on update of hook where you used this state as internal move line.
Having a state for each is cleaner I think